src/main/java/org/jabref/logic/ai/chatting/AiChatLogic.java

```diff
@@ -48,21 +46,23 @@ public class AiChatLogic {
     private final ChatLanguageModel chatLanguageModel;
     private final EmbeddingModel embeddingModel;
     private final EmbeddingStore<TextSegment> embeddingStore;
-    private final Executor cachedThreadPool;
+    private final TemplatesService templatesService;
 
     private final ObservableList<ChatMessage> chatHistory;
     private final ObservableList<BibEntry> entries;
     private final StringProperty name;
     private final BibDatabaseContext bibDatabaseContext;
 
     private ChatMemory chatMemory;
-    private Chain<String, String> chain;
+
+    // You must not pass an empty list to langchain4j {@link IsIn} filter. We forced to use null here, instead of Optional.
+    private @Nullable Filter filter;
```

Comment:
Normally, one does "good" Java code on the own side - and transformation to a "bad" API when calling.

E.g.:  `filter.isEmpty() ? null : filter`